core/task/manager.go

```diff
@@ -98,28 +102,92 @@ func (m*Manager) NewTaskForMesosOffer(offer *mesos.Offer, descriptor *Descriptor
 	return
 }
 
-func (m *Manager) RefreshClasses() (err error) {
+// TODO: Any reason to be a *Manager method? If not then should it reside here?
+func getTaskClassList(taskClassesRequired []string) (taskClassList []*TaskClass, err error) {
+	repoManager := the.RepoManager()
+	var yamlData []byte
+
+	taskClassList = make([]*TaskClass, 0)
+
+	for _, taskClass := range taskClassesRequired {
+		taskClassString := strings.Split(taskClass, "@")
+		taskClassFile := taskClassString[0] + ".yaml"
+		var repo *repos.Repo
+		repo, err = repos.NewRepo(strings.Split(taskClassFile, "tasks/")[0])
+		if err != nil {
+			return
+		}
+		repo = repoManager.GetRepos()[repo.GetIdentifier()] //get repo pointer from repomanager
+	 	if repo == nil { //should never end up here
+			return nil, errors.New("getTaskClassList: repo not found for " + taskClass)
+		}
+
+		yamlData, err = ioutil.ReadFile(viper.GetString("repositoriesUri") + taskClassFile)
+		if err != nil {
+			return nil, err
+		}
+		//var taskClass *TaskClass //TODO: This doesn't unmarshal; unclear why
```

Comment:
Let's figure this out together.

Reply:
No luck with this one.

Reply:
Now I know why: if you look at a task class file, for example https://gitlab.cern.ch/tmrnjava/aliecs_conf/blob/master/tasks/readout.yaml you can notice that there's a `-` in front of everything. Translating that to JSON, the file contains a structure like `[{'name':'readout', ...}]` rather than `{'name':'readout', ...}`.
Since we expect to have one and only one task class descriptor per task class file, the file should be modified as follows, so that the top level node is the dict itself and not an array:
```
name: readout # read by workflow
control:
  mode: direct
wants:
  cpu: 0.15
  memory: 128
bind: []
properties: {}
command: 
  env: []
  shell: true
  arguments:
    - "file:/opt/alisw/el7/Readout/v0.24-occ-1/etc/readout.cfg"
  value: >-
    source /etc/profile.d/modules.sh &&
    MODULEPATH=/opt/alisw/el7/modulefiles module load Readout &&
    MODULEPATH=/opt/alisw/el7/modulefiles module load Control-OCCPlugin &&
    MODULEPATH=/opt/alisw/el7/modulefiles module list &&
    readout.exe
```
Then you'll be able to unmarshal into the *TaskClass directly instead of the slice.

Reply:
I've taken the liberty of porting the sample tasks and workflows from `config.yaml` to this repo https://gitlab.cern.ch/tmrnjava/aliecs_conf so feel free to use them when testing this. Generally once we can load the FairMQ sample workflows, we can trust everything is in place to merge.